src/Microsoft.AspNetCore.Sockets.Client/WebSocketsTransport.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO.Pipelines;
+using System.Net.WebSockets;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.Sockets.Client
+{
+    public class WebSocketsTransport : ITransport
+    {
+        private ClientWebSocket _webSocket = new ClientWebSocket();
+        private IChannelConnection<Message> _application;
+        private CancellationToken _cancellationToken = new CancellationToken();
+
+        public Task Running { get; private set; }
+                                                
+        public async Task StartAsync(Uri url, IChannelConnection<Message> application)  
+        {
+            if (url == null)
+            {
+                throw new ArgumentNullException(nameof(url));
+            }
+            if (application == null)
+            {
+                throw new ArgumentNullException(nameof(application));
+            }
+            _application = application;
+            var webSocketUrl = url.ToString();
+            await Connect(webSocketUrl);
+            var sendTask = SendMessages(url, _cancellationToken);
+            var receiveTask = ReceiveMessages(url, _cancellationToken);
+
+            Running = Task.WhenAll(sendTask, receiveTask).ContinueWith(t => {
+                _application.Output.TryComplete(t.IsFaulted ? t.Exception.InnerException : null);
+                return t;
+            }).Unwrap();
+        }
+
+        public async Task ReceiveMessages(Uri pollUrl, CancellationToken cancellationToken)
+        {
+            var totalBytes = 0;
+            var incomingMessage = new List<byte[]>();
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                bool completedMessage;
+                WebSocketReceiveResult receiveResult;
+                ArraySegment<byte> buffer;
+                do
+                {
+                    buffer = new ArraySegment<byte>(new byte[1024]);
+                    receiveResult = await _webSocket.ReceiveAsync(buffer, cancellationToken);
+                    completedMessage = receiveResult.EndOfMessage;
+                    incomingMessage.Add(buffer.Array);
+                    totalBytes += buffer.Count;
```

Comment:
This is the `Count` from the `ArraySegment` you passed in to `ReceiveAsync`, which will always be `1024` since you created it from the array above (`ReceiveAsync` doesn't modify it, since it's passed in by-value). The actual number of bytes read into the buffer is available in `receiveResult`.